packages/nuxt/src/app/composables/cookie.ts

```diff
@@ -35,8 +35,24 @@ export function useCookie<T = string | null | undefined> (name: string, _opts?:
 
   if (process.client) {
     const callback = () => { writeClientCookie(name, cookie.value, opts as CookieSerializeOptions) }
+
     if (opts.watch) {
-      watch(cookie, callback, { deep: opts.watch !== 'shallow' })
+      const channel = new BroadcastChannel(`nuxt:cookies:${name}`)
+      onUnmounted(() => { channel.close() })
+      let watchPaused = false
+
+      watch(cookie, (newValue) => {
+        if (watchPaused) { return }
```

Comment:
shall we also skip if the new value is the same as old, to avoid infinite ping-pong?

Reply:
But the watch is not triggered if value is the same, as far as I know.

Reply:
```js
    channel.onmessage = (event) => {
      if (isEqual(cookie.value, event.data)) { return }
      watchPaused = true
      cookie.value = event.data
      nextTick(() => { watchPaused = false })
    }
```

Something like this?

Reply:
Or i could put it inside the watcher